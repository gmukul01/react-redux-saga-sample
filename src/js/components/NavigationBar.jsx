import React, { Component, Fragment } from "react";
import { Link } from "react-router-dom";
import history from "util/history";
import classNames from "classnames";

export default class NavigationBar extends Component {
  render() {
    const pathname = history.location.pathname,
      homeClass = classNames({ "link--active": pathname === "/" }),
      catalogueClass = classNames({ "link--active": pathname === "/catalogues" }),
    return (
      <nav className="navbar">
        <Link to="/" className={homeClass}>
          Home
        </Link>
        <Link to="/catalogues" className={catalogueClass}>
          Catalogues
        </Link>
      </nav>
    );
  }
}
