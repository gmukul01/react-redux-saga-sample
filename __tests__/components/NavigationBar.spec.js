import React from "react";
import sinon from "sinon";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";

import history from "util/history";
import NavigationBar from "components/NavigationBar";

describe("Teams page", () => {
  let stub = sinon.stub(history.location, "pathname");

  afterEach(() => {
    stub.restore();
  });

  it("should render correctly", () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<NavigationBar />);
    expect(result).toMatchSnapshot();
  });

  it("should show home link as active when pathname is '/'", () => {
    stub.value("/");
    const wrapper = shallow(<NavigationBar />);
    expect(wrapper.find({ to: "/" }).prop("className")).toEqual("link--active");
  });

  it("should show home link as active when pathname is '/catalogues'", () => {
    stub.value("/catalogues");
    const wrapper = shallow(<NavigationBar />);
    expect(wrapper.find({ to: "/catalogues" }).prop("className")).toEqual("link--active");
  });
});
