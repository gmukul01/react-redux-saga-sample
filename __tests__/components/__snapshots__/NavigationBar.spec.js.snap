// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teams page should render correctly 1`] = `
<nav
  className="navbar"
>
  <Link
    className="link--active"
    replace={false}
    to="/"
  >
    Home
  </Link>
  <Link
    className=""
    replace={false}
    to="/catalogues"
  >
    Catalogues
  </Link>
</nav>
`;
